mm2src/coins/utxo.rs

```diff
@@ -2647,5 +2663,109 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     }
     // next 2 lines ported as is from Komodo codebase
     minutes -= 59;
-    (value / 10_512_000) * minutes
+    let accrued = (value / 10_512_000) * minutes;
+
+    Ok(accrued)
+}
+
+fn kmd_interest_accrue_stop_at(height: u64, lock_time: u64) -> u64 {
+    let seconds = if height < 1_000_000 {
+        // interest stop accruing after 1 year before block 1000000
+        365 * 24 * 60 * 60
+    } else {
+        // interest stop accruing after 1 month past 1000000 block
+        31 * 24 * 60 * 60
+    };
+
+    lock_time + seconds
+}
+
+fn kmd_interest_accrue_start_at(lock_time: u64) -> u64 {
+    let one_hour = 60 * 60;
+    lock_time + one_hour
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+enum KmdRewardsNotAccruedReason {
+    LocktimeNotSet,
+    LocktimeLessThanThreshold,
+    UtxoHeightGreaterThanEndOfEra,
+    UtxoAmountLessThanTen,
+    OneHourNotPassedYet,
+    TransactionInMempool,
+}
+
+#[derive(Serialize)]
+enum KmdRewardsAccrueInfo {
+    Accrued(BigDecimal),
+    NotAccruedReason(KmdRewardsNotAccruedReason),
+}
+
+#[derive(Serialize)]
+pub struct KmdRewardsInfoElement {
+    tx_hash: H256Json,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    height: Option<u64>,
+    /// The zero-based index of the output in the transaction’s list of outputs.
+    output_index: u32,
+    amount: BigDecimal,
+    locktime: u64,
+    /// Amount of accrued rewards.
+    accrued_rewards: KmdRewardsAccrueInfo,
+    /// Rewards start to accrue at this time for the given transaction.
+    accrue_start_at: u64,
```

Comment:
I agree with Tony at this point. `accrue_start_at` and `accrue_stop_at` should not be in response if rewards can not be accrued at all. But there are cases when rewards are not accruing yet, but they will be accrued, so it seems like we should duplicate this info for some enum variants, so it will possibly look like this:
```
enum KmdRewardsNotAccruedReason {
OneHourNotPassedYet { accrue_start_at: u64, accrue_stop_at: u64 },
TransactionInMempool { accrue_start_at: u64, accrue_stop_at: u64 },
...
}
```
And 
```
enum KmdRewardsAccrueInfo { 
Accrued { accrued: BigDecimal, accrue_start_at: u64, accrue_stop_at: u64 }
...
}
```

This is really last change request :) Thanks a lot!

Reply:
The requested changes were implemented in the commit above, thanks!